src/tests/oracle.rs

```diff
@@ -2404,32 +2411,75 @@ fn update_rewards() {
 				expected_accumulated_reward_per_share
 			);
-			// todo:
-			// let expected_staking_rewards_balance = U256ToBalance::convert(
-			// 	Amount::from(staking_rewards) -
-			// 		(Amount::from(expected_accumulated_reward_per_share) *
-			// 			Amount::from(token(100))),
-			// );
-			//assert_eq!(Tellor::staking_rewards_balance(), expected_staking_rewards_balance);
-			timestamp
+
+			let staking_rewards_balance = Balances::free_balance(staking_rewards_account);
+			let staked_balance = Tellor::convert(staker_info.staked_balance).unwrap();
+			let expected_staking_rewards_balance = U256ToBalance::convert(
+				U256::from(staking_rewards) -
+					(staked_balance * U256::from(previous_accumulated_reward_per_share) /
+						U256::from(unit) - U256::from(staker_info.reward_debt)),
+			);
+			assert_eq!(staking_rewards_balance, expected_staking_rewards_balance);
+
+			(timestamp, expected_staking_rewards_balance)
 		});
 
 		// advance time 30 days
-		let (timestamp_4, expected_accumulated_reward_per_share) =
+		let (timestamp_4, expected_accumulated_reward_per_share, expected_staking_rewards_balance) =
 			with_block_after(86_400 * 30, || {
+				let accumulated_reward_per_share = Tellor::accumulated_reward_per_share();
+				let reward_rate = Tellor::reward_rate();
+
 				// update rewards
 				assert_ok!(Tellor::update_rewards());
 				let timestamp = now();
 
 				assert_eq!(timestamp, timestamp_3 + 86_400 * 30 + 1);
 				assert_eq!(Tellor::time_of_last_allocation(), timestamp);
 				assert_eq!(Tellor::reward_rate(), 0); // rewards ran out, reward rate should be 0
-				let expected_accumulated_reward_per_share = token(1000) / 100;
+
+				let new_accumulated_reward_per_share = U256::from(accumulated_reward_per_share) +
+					(U256::from(timestamp - timestamp_3) *
+						U256::from(reward_rate) * U256::from(unit)) /
+						U256::from(Tellor::get_total_stake_amount());
+
+				let accumulated_reward = (new_accumulated_reward_per_share *
+					U256::from(Tellor::get_total_stake_amount())) /
+					U256::from(unit) - U256::from(Tellor::total_reward_debt());
+
+				assert!(accumulated_reward >= U256::from(expected_staking_rewards_balance));
+
+				// Calculation of new_pending_rewards based on actual update_rewards() implementation
+
+				/*let new_pending_rewards =  U256::from(staking_rewards_balance) -
+					(U256::from(Tellor::accumulated_reward_per_share()) *
+						U256::from(Tellor::get_total_stake_amount()) /
+						U256::from(unit) -
+						U256::from(Tellor::total_reward_debt())
+					);
+				let expected_accumulated_reward_per_share1 = Tellor::accumulated_reward_per_share() +
+					U256ToBalance::convert((new_pending_rewards * U256::from(unit)) / U256::from(Tellor::get_total_stake_amount()));
+				*/
+
+				// Calculation of new_pending_rewards based tellorFlex's test case.
+				// Question: Why we are using hard-coded (150 * unit) instead of using actual business logic for calculating new_pending_rewards?
```

Comment:
We are using hardcoded or expected values in order to attempt to duplicate the expected logic, and catch any errors in implementation. See this PR for clarification and calculating using both methods: https://github.com/tellor-io/tellorFlex/pull/76